hoomd/BoxDim.h

```diff
@@ -393,6 +452,120 @@ struct
         return vec3<Scalar>(minImage(vec_to_scalar3(v)));
         }
 
+    //! Compute minimum image
+    /*!
+        \param dr Position vector to compute / Minimum image vector returned
+        \param dv Velocity vector to compute / Minimum image vector returned
+        \note \a dr must not extend more than 1 image beyond the box
+    */
+    HOSTDEVICE void minImage(Scalar3& dr, Scalar3& dv) const
+        {
+        Scalar3 r = dr;
+        Scalar3 v = dv;
+        Scalar3 L = getL();
+
+#ifdef __HIPCC__
+        if (m_periodic.z)
+            {
+            Scalar img = slow::rint(r.z * m_Linv.z);
+            r.z -= L.z * img;
+            r.y -= L.z * m_yz * img;
+            r.x -= L.z * m_xz * img;
+            v.z -= m_L_rate.z * img;
+            v.y -= L.z * m_yz_rate * img;
+            v.x -= L.z * m_xz_rate * img;
+            }
+
+        if (m_periodic.y)
+            {
+            Scalar img = slow::rint(r.y * m_Linv.y);
+            r.y -= L.y * img;
+            r.x -= L.y * m_xy * img;
+            v.y -= m_L_rate.y * img;
+            v.x -= L.y * m_xy_rate * img;
+            }
+
+        if (m_periodic.x)
+            {
+            Scalar img = slow::rint(r.x * m_Linv.x);
+            r.x -= L.x * img;
+            v.x -= m_L_rate.x * img;
+            }
+#else
+        // on the cpu, branches are faster than calling rint
+        if (m_periodic.z)
+            {
+            if (r.z >= m_hi.z)
+                {
+                r.z -= L.z;
+                r.y -= L.z * m_yz;
+                r.x -= L.z * m_xz;
+                v.z -= m_L_rate.z;
+                v.y -= L.z * m_yz_rate;
+                v.x -= L.z * m_xz_rate;
+                }
+            else if (r.z < m_lo.z)
+                {
+                r.z += L.z;
+                r.y += L.z * m_yz;
+                r.x += L.z * m_xz;
+                v.z += m_L_rate.z;
+                v.y += L.z * m_yz_rate;
+                v.x += L.z * m_xz_rate;
+                }
+            }
+
+        if (m_periodic.y)
+            {
+            if (r.y >= m_hi.y)
+                {
+                int i = int(r.y * m_Linv.y + Scalar(0.5));
```

Comment:
This is what's called a C-style cast, but this way is preferred now:
```suggestion
                const int i = static_cast<int>(r.y * m_Linv.y + Scalar(0.5));
```